feflow/tests/test_protein_mutation.py

```diff
@@ -476,3 +525,71 @@ def test_charge_changing_convergence(
             f"DDG ({arg_lys_diff}) is greater than "
             f"6 * dDDG ({6 * arg_lys_diff_error})"
         )
+
+    def test_proline_mutation_fails(
+        self, ala_capped_system, pro_capped_system, ala_to_pro_mapping
+    ):
+        """Test that attempting to make a protein mutation that involves proline (or ring breaking
+        transformations) is not handled and results in an error.
+
+        This test ensures that the mutation protocol correctly identifies mutations involving
+        proline, which is typically a ring-breaking transformation that is not supported, and
+        raises an appropriate error.
+
+        Parameters
+        ----------
+        ala_capped_system : ChemicalSystem
+            The chemical system representing a capped alanine residue.
+        pro_capped_system : ChemicalSystem
+            The chemical system representing a capped proline residue.
+        ala_to_pro_mapping : LigandAtomMapping
+            Mapping object representing the atom mapping from ALA to PRO.
+        """
+        from feflow.utils.exceptions import MethodConstraintError
+
+        settings = ProteinMutationProtocol.default_settings()
+        protocol = ProteinMutationProtocol(settings=settings)
+
+        # Expect an error when trying to create the DAG with this invalid transformation
+        with pytest.raises(MethodConstraintError, match="proline.*not supported"):
```

Comment:
I am unsure if it's a valid use case, but do you intend to stop all mutations involving proline? So, a proline-to-proline mutation would not be allowed in very different conformations as well?